libs/features/mas/web-components/src/aem.js

```diff
@@ -1,63 +1,211 @@
+            Authorization: `Bearer ${sessionStorage.getItem('masAccessToken') ?? window.adobeid?.authorize?.()}`,
+            pragma: 'no-cache',
+            'cache-control': 'no-cache',
         };
     }
-    const searchParams = new URLSearchParams({
-        query: JSON.stringify({ filter }),
-    }).toString();
-    return fetch(`${this.cfSearchUrl}?${searchParams}`, {
-        headers,
-    })
-        .then((res) => res.json())
-        .then(({ items }) => items);
-}
 
-async function getCfByPath(path) {
-    return fetch(`${this.cfFragmentsUrl}?path=${path}`, {
-        headers,
-    })
-        .then((res) => res.json())
-        .then(({ items: [item] }) => item);
-}
+    async getCsrfToken() {
+        const { token } = await fetch(this.csrfTokenUrl, {
+            headers: this.headers,
+        }).then((res) => res.json());
+        return token;
+    }
+
+    /**
+     * Search for content fragments
+     * @param {Object} params - The search options
+     * @param {string} [params.path] - The path to search in
+     * @param {string} [params.query] - The search query
+     * @param {string} [params.variant] - The variant to filter by
+     * @returns {Promise<Array>} - A promise that resolves to an array of search results
+     */
+    async searchFragment({ path, query, variant }) {
+        const filter = {};
+        if (path) {
+            filter.path = path;
+        }
+        if (query) {
+            filter.fullText = {
+                text: encodeURIComponent(query),
+                queryMode: 'EXACT_WORDS',
+            };
+        }
+
+        const searchParams = new URLSearchParams({
+            query: JSON.stringify({ filter }),
+        }).toString();
+        return fetch(`${this.cfSearchUrl}?${searchParams}`, {
+            headers: this.headers,
+        })
+            .then((res) => res.json())
+            .then((json) => json.items)
+            .then((items) => {
+                if (variant) {
+                    return items.filter((item) => {
+                        const [itemVariant] = item.fields.find(
+                            (field) => field.name === 'variant',
+                        )?.values;
+                        return itemVariant === variant;
+                    });
+                }
+                return items;
+            });
+    }
+
+    /**
+     * Get fragment by path
+     * @param {string} path fragment path
+     * @returns {Promise<Object>} the raw fragment item
+     */
+    async getFragmentByPath(path) {
+        const headers = this.#author ? this.headers : {};
+        return fetch(`${this.cfFragmentsUrl}?path=${path}`, { headers })
+            .then((res) => res.json())
+            .then(({ items: [item] }) => item);
+    }
+
+    async getFragment(res) {
+        const eTag = res.headers.get('Etag');
+        const fragment = await res.json();
+        fragment.etag = eTag;
+        return fragment;
+    }
```

Comment:
Should we add a check for the response in here?
```suggestion
    async getFragment(res) {
    if(res.ok) {
          const eTag = res.headers.get('Etag');
          const fragment = await res.json();
          fragment.etag = eTag;
          return fragment;
      } 
       throw new Error('Failed to get fragment by id');
    }
```